### PR TITLE
[Azure Search] Introduce a few breaking changes in swagger specifications

### PR DESCRIPTION
**Pull request by arv100kri**
This PR attempts at bringing the swagger spec in line with the REST API behavior, which intentionally introduces a few breaking changes (which will follow in a subsequent major version SDK update)

- Skills can now optionally take in a `name` property
- WebApiSkill has an optional `degreeOfParallelism` property, which can allow customers to specify how many API calls (if non-null, non-zero) can be made in parallel to their endpoint
- Add more information to indexer execution's errors and warnings

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/search/data-plane/Microsoft.Azure.Search.Service/stable/2019-05-06/examples/SearchServiceCreateOrUpdateSkillset.json [hunk 1]
```diff
@@ -84,6 +84,31 @@
               "targetName": "keyPhrases"
             }
           ]
+        },
+        {
+          "@odata.type": "#Microsoft.Skills.Custom.WebApiSkill",
```
Comment (brjohnstmsft) [inline specification/search/data-plane/Microsoft.Azure.Search.Service/stable/2019-05-06/examples/SearchServiceCreateOrUpdateSkillset.json]: Is `Custom` really part of this namespace? Was it always that way? Are there any other classes in the `Custom` namespace?

File: specification/search/data-plane/Microsoft.Azure.Search.Service/stable/2019-05-06/examples/SearchServiceGetIndexerStatus.json [hunk 2]
```diff
@@ -43,17 +43,38 @@
...
+                "name": null,
+                "details": null,
+                "documentationLink": null
+              },
+              {
+                "key": "document id 2",
+                "message": "Another interesting scenario happened.",
+                "name": "Enrichment.LanguageDetectionSkill.#4",
+                "details": "The skill did something that didn't break anything, nonetheless something we didn't expect happened, so it might be worth double checking.",
+                "documentationLink": "https://contoso.example.org?type=warning"
```
Comment (brjohnstmsft) [inline specification/search/data-plane/Microsoft.Azure.Search.Service/stable/2019-05-06/examples/SearchServiceGetIndexerStatus.json]: From this example, it looks as if the customer somehow provides the `documentationLink` via the custom Web API skill. Is that the case, or are these links supposed to be generated internally by us? If it's the latter, the examples should not use the contoso host/domain.

File: specification/search/data-plane/Microsoft.Azure.Search.Service/stable/2019-05-06/examples/SearchServiceListSkillsets.json [hunk 3]
```diff
@@ -14,6 +14,7 @@
             "skills": [
               {
                 "@odata.type": "#Microsoft.Skills.Text.EntityRecognitionSkill",
+                "name": "#1",
```
Comment (brjohnstmsft) [inline specification/search/data-plane/Microsoft.Azure.Search.Service/stable/2019-05-06/examples/SearchServiceListSkillsets.json]: Are these meant to be examples of explicitly-set names or implicitly-set names? If the latter, the docs for the name property don't indicate that the name will be prefaced with "#". If the former, the example should be something other than a number so the two cases aren't confused.

**Review by bitsforthought — CHANGES_REQUESTED** [review]

**Comment by brjohnstmsft** [comment]
@arv100kri Please ensure that the example changes pass validation (not sure at a glance that they're failing validation because there are other known issues with validating our Swagger right now).

File: specification/search/data-plane/Microsoft.Azure.Search.Service/stable/2019-05-06/examples/SearchServiceCreateOrUpdateSkillset.json (hunk 1 above)
Comment (arv100kri) [reply specification/search/data-plane/Microsoft.Azure.Search.Service/stable/2019-05-06/examples/SearchServiceCreateOrUpdateSkillset.json]: Yup, this was always this way and so far there is only this one skill in this namespace

File: specification/search/data-plane/Microsoft.Azure.Search.Service/stable/2019-05-06/examples/SearchServiceListSkillsets.json (hunk 3 above)
Comment (arv100kri) [reply specification/search/data-plane/Microsoft.Azure.Search.Service/stable/2019-05-06/examples/SearchServiceListSkillsets.json]: This is an implicitly-set name - I'll update the docs to mention that!

File: specification/search/data-plane/Microsoft.Azure.Search.Service/stable/2019-05-06/examples/SearchServiceGetIndexerStatus.json (hunk 2 above)
Comment (arv100kri) [reply specification/search/data-plane/Microsoft.Azure.Search.Service/stable/2019-05-06/examples/SearchServiceGetIndexerStatus.json]: The latter, I'll use an actual link that we'll send (it's still going to be a forward link)

**Comment by arv100kri** [comment]
> @arv100kri Please ensure that the example changes pass validation (not sure at a glance that they're failing validation because there are other known issues with validating our Swagger right now).

It's the same "base" set of validations errors for the various enum types

File: specification/search/data-plane/Microsoft.Azure.Search.Service/stable/2019-05-06/searchservice.json [hunk 4]
```diff
@@ -4578,7 +4578,7 @@
         },
         "name": {
           "type": "string",
-          "description": "The name of the skill which uniquely identifies it within the skillset. A skill with no name defined will be given a default name of its 1-based index in the skills array."
+          "description": "The name of the skill which uniquely identifies it within the skillset. A skill with no name defined will be given a default name of its 1-based index in the skills array, prefixed with the character '#'."
```
Comment (brjohnstmsft) [inline specification/search/data-plane/Microsoft.Azure.Search.Service/stable/2019-05-06/searchservice.json]: Much better :)

**Review by bitsforthought — APPROVED** [review]

**Comment by yaohaizh** [comment]
@arv100kri   The model validation failed with https://dev.azure.com/azure-sdk/public/_build/results?buildId=90022&view=logs&j=8d2f9c49-cd83-5a9d-f3fe-2fd651afa742&t=43324f1f-6724-5d61-785f-f0ac9fd256a1&l=272 Please help fix it. Meanwhile, we need Azure API board review this. Thanks.

**Comment by brjohnstmsft** [comment]
@yaohaizh Those validation failures are the result of a known issue. See this for context: https://github.com/Azure/azure-rest-api-specs/pull/5665#issuecomment-514339350

Regarding Azure API review board -- This PR is a change that brings our Swagger in line with the API that's actually deployed in production. Is a review still required in this case?

**Comment by yaohaizh** [comment]
As talked with @brjohnstmsft , the change is already deployed and the change reflect the service deployed.  Given this, @yungezz , could you help merge this since it is blocked by model validation.

**Review by yaohaizh — APPROVED** [review]

**Comment by yungezz** [comment]
checked with @yaohaizh, this PR is ready for merge. ModelValidation error is an known issue.

**Pull request merged**